registration-processor/core-processor/registration-processor-abis-handler-stage/src/main/java/io/mosip/registration/processor/abis/handler/stage/AbisHandlerStage.java

```diff
@@ -599,6 +608,32 @@ public String getFilename() {
 
 		return response.getDataShare().getUrl();
 	}
+	private void validateBiometricRecord(BiometricRecord biometricRecord, List<String> modalities) throws DataShareException {
+		if(biometricRecord == null  || biometricRecord.getSegments() == null
+				|| biometricRecord.getSegments().isEmpty() || modalities==null || modalities.isEmpty()) {
+			throw new DataShareException("No Biometrics Found with Data Share Policy");
+
+		}
+		
+		boolean flag = false;
+		for(BIR bir : biometricRecord.getSegments()) {

```

Comment:
this loop and check has to be otherway around. For each modalities, we should have that info in  the biometric record if not exception.

Reply:
But we discussed like atleast 1 has to be present

Reply:
Incorporated review comment